### PR TITLE
docs: remove chat:write.public scope for best practice

### PR DESCRIPTION
**Pull request by mwbrooks**
### Summary

This pull request removes references to the `chat:write.public` scope to encourage best practices since this scope is more permissive and should be used sparingly.

This change was originally proposed by @GPTI314 in PR https://github.com/slackapi/slack-cli/pull/267. Since the PR is inactive and the CLA is not signed, I've re-written it in this PR.

* Related to https://github.com/slackapi/slack-cli/pull/267
* Closes https://github.com/slackapi/slack-cli/pull/267

### Requirements

* [x] I've read and understood the [Contributing Guidelines](/blob/main/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).

### CODE REVIEW

File: internal/goutils/strings_test.go [hunk 1]
```diff
@@ -326,8 +326,8 @@ func Test_RedactPII(t *testing.T) {
 		},
 		{
 			name:     "Escape sensitive data from mock HTTP response",
-			text:     `{"ok":true,"app_id":"A123","credentials":{"client_id":"123","client_secret":"123","verification_token":"123","signing_secret":"123"},"oauth_authorize_url":"123":\/\/slack.com\/oauth\/v2\/authorize?client_id=123&scope=commands,chat:write,chat:write.public"}`,
-			expected: `{"ok":true,"app_id":"A123","credentials":{"client_id":"...","client_secret":"...","verification_token":"...","signing_secret":"..."},"oauth_authorize_url":"...":\/\/slack.com\/oauth\/v2\/authorize?client_id=...&scope=commands,chat:write,chat:write.public"}`,
```
Comment (zimeg) [inline internal/goutils/strings_test.go:330]: 🧪 thought: This might reduce the quality of these tests in some perhaps odd edge cases but let's not let that block these greps!

**Review by zimeg — APPROVED** [review]
@mwbrooks Thanks for the polish to these pages! 🔮 ✨ 

Let's merge this and save time for revisiting these pages for unrelated changes soon. I noted some testing concerns, but nothing that I'll stress about in future weeks. I hope.
